### ISSUE TITLE
Docker Desktop 4.8.2 not working after updating to macOS Monterey 12.4

### ISSUE THREAD

**Issue by dspacejs** [docker/for-mac#6331]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 00761D89-90BD-4FBB-ABF4-080F56FA00D7/20220529080222
  

### Expected behavior

Docker should start without errors.

### Actual behavior

It's not working at all and will not start, despite trying many things. This is the output of `docker ps`:

```
Error response from daemon: dial unix /Users/daniel/Library/Containers/com.docker.docker/Data/docker.raw.sock: connect: no such file or directory
```

### Information

  - macOS Version: 12.4
  - Intel chip or Apple chip: Apple
  - Docker Desktop Version: 4.8.2 (79419)
  
The app has not working at all since updating to macOS Monterey 12.4 today. It was working fine before then.

I've tried the following:

1. Clicking any of the troubleshooting buttons, such as these does nothing. It's completely unresponsive:

![Screen Shot 2022-05-29 at 5 34 39 pm](https://user-images.githubusercontent.com/9282837/170858473-7ce9bf2b-110c-4d32-8bc1-4bc86597f011.png)

2. Reinstalling `docker` with `brew` doesn't work:

```sh
brew uninstall --cask docker
brew install --cask docker
```

3. Uninstalling the app from Brew, and then installing the [Apple Silicon .dmg from your website](https://www.docker.com/products/docker-desktop/) didn't work.

4. Removing the app's files and then reinstalling via step 1 or 2 didn't work:

```sh
rm -rf ~/.docker
rm -rf ~/Library/Saved Application State/com.electron.docker-frontend.savedState
rm -rf ~/Library/Preferences/com.electron.docker-frontend.plist
rm -rf ~/Library/Preferences/com.docker.docker.plist
rm -rf ~/Library/Group\ Containers/group.com.docker/
rm -rf ~/Library/Cookies/com.docker.docker.binarycookies
rm -rf ~/Library/Caches/com.docker.docker
rm -rf ~/Library/Application\ Support/Docker\ Desktop/
```

 5. Restarting my Mac doesn't work.
 6. Restarting the Docker app doesn't work.
 7. Killing all Docker processes with the Activity Monitor doesn't work.
 8. Deleting the app from `/Applications` and then reinstalling it doesn't work.

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

[diagnostics.log](https://github.com/docker/for-mac/files/8792879/diagnostics.log)

### Steps to reproduce the behavior

I'm not sure how to reproduce this.

**Comment by phamnhungoctuan** [comment]
same problem :( 

Macbook M1 12.4

**Comment by dspacejs** [comment]
@phamnhungoctuan did you have any luck troubleshooting or fixing it?

**Comment by phamnhungoctuan** [comment]
> @phamnhungoctuan did you have any luck troubleshooting or fixing it?

I have fixed it. Plz follow this https://stackoverflow.com/a/65468254

**Comment by dspacejs** [comment]
@phamnhungoctuan thanks, unfortunately I still have the problem, though. It's stuck on the "Docker Desktop starting..." step:

![Screen Shot 2022-05-30 at 5 47 29 pm](https://user-images.githubusercontent.com/9282837/170948903-e123c9aa-d073-4bc3-a190-b26f5df75428.png)


I have no idea why it's so broken on my device now. I may just have to wait for one of the Docker engineers to respond at this point.

**Comment by chuck-g** [comment]
I think this is a duplicate of [6324](https://github.com/docker/for-mac/issues/6324)

**Comment by robsontenorio** [comment]
Same problem on M1 Mac with macOS 12.4 (21F79).

Just fully uninstalled it, and installed with brew but no lucky. I have tried install 4.8.1, but same problem.

Any tips?

**Comment by ulises-castro** [comment]
Same here... (I updated to 12.4 yesterday and now it does not work anymore)

I even tried to using rancher desktop instead of docker desktop but it does not work neither...

**Comment by robsontenorio** [comment]
Found a workaround. Its working.

https://forums.docker.com/t/docker-desktop-4-6-1-for-mac-wont-start/122605/9

![image](https://user-images.githubusercontent.com/118955/171683943-f4697b1f-ca39-4a89-b4cb-afe46367d616.png)


After that,  just go through preference and reset do factory defaults.

![image](https://user-images.githubusercontent.com/118955/171684094-a18503b4-14ea-4800-ad44-845eac085e20.png)

**Comment by fredericdalleau** [comment]
@robsontenorio did you have a shared directory in the list that no longer existed?

**Comment by robsontenorio** [comment]
@fredericdalleau I am not sure anymore, because tried too many things to make it work. Problem started when upgrade to macOS 12.4, as many people related. 

Just followed workaround above and worked. Hopping someone can confirm.

**Comment by robsontenorio** [comment]
@ulises-castro you can also try remove this. 

```

rm -rf Library/Group\ Containers/group.com.docker
rm -rf Library/Containers/com.docker.docker
rm -rf Library/Application\ Support/Docker\ Desktop

**Comment by sneakerhax** [comment]
@robsontenorio tried the workaround and the removing of the suggested files but it's still not working for me.

**Comment by robsontenorio** [comment]
@sneakerhax it worth mentioning I am now on macOS 12.5 beta. 

But I am not sure what step exactly solved.

**Comment by maxeth** [comment]
> @ulises-castro you can also try remove this.

After I do that my Docker Desktop App doesn't even start. Did anyone find any other working solution?

**Comment by maxeth** [comment]
> Found a workaround. Its working.
> 
> https://forums.docker.com/t/docker-desktop-4-6-1-for-mac-wont-start/122605/9
> 
> ![image](https://user-images.githubusercontent.com/118955/171683943-f4697b1f-ca39-4a89-b4cb-afe46367d616.png)
> 
> After that Docker starts. Then just go through preference and reset do factory defaults.
> 
> ![image](https://user-images.githubusercontent.com/118955/171684094-a18503b4-14ea-4800-ad44-845eac085e20.png)
> 
> Finally, enable VirtioFS again
> 
> ![image](https://user-images.githubusercontent.com/118955/171685221-9729c70e-b6d5-4817-b56a-06a665a2805c.png)

After I do that my Docker Desktop App doesn't even start. Did anyone find any other working solution?

**Comment by maxeth** [comment]
@fredericdalleau  Hello, the OP posted a Diagnostics ID (```Diagnostics ID: 00761D89-90BD-4FBB-ABF4-080F56FA00D7/20220529080222```). I'm experiencing the exact same issue with the same error message on macOS Monterey 12.4, so I don't think it's necessary to create a report on my machine, but I could quickly do that as well if needed 🙂

**Comment by fredericdalleau** [comment]
Hello again, the issue #6324 was similar and the author reported DD 4.9 fixed it for him, give it a try!

**Comment by sneakerhax** [comment]
> Hello again, the issue #6324 was similar and the author reported Docker Desktop 4.9 fixed it for him, give it a try!

I did a fresh install of DD 4.9 and this did not fix my issues.

**Comment by dspacejs** [comment]
Upgrading to v4.9.0 fixed it to me. However it looks like it hasn't for @sneakerhax (and probably others), so I'll leave this issue open.

**Comment by sneakerhax** [comment]
> I've been looking at @sneakerhax diag (thanks btw), it could be a firewall issue. Can you check your firewall settings in "Settings > Security and Privacy > Firewall"?

I reviewed the firewall setting and they're set to off

I have worked through a few troubleshooting steps with support (I have Docker Pro). 

My new Diagnostic ID: E40EA4D1-CC34-47AB-B833-44F2C5A07FF9/20220607214757

The issue was not yet fixed by any of the steps provided by support

**Comment by maxeth** [comment]
> Update: With the additional context provided to me by support about Docker attempting to connect to localhost I tried disabling all my VPN adapters (work and personal) and the Docker desktop now boots. Even when connected to VPN after disabling and re-enabling the VPN adapters it now works!

I can also confirm that disabling my VPN before starting Docker Desktop fixed the issue. Still, I don't think I had the same issue before macOS 12.4, but I'm not sure.

**Comment by ulises-castro** [comment]
Hey guys, just a quick update about my case.

I installed docker's 4.9.0 version and deactivated my VPN and now docker seems be working fine, idk what was change on 12.4 MacOs' update, but it used to work along my vpn, I had to desactive my vpn then open docker desktop and activate my vpn again.

**Comment by zcht** [comment]
Unfortunately have the same problem: today updated to Monterey 12.4, DD 4.8.2 could no longer be started. Error code 2 - DD is simply closed, reset to factory settings has brought nothing. Update to DD 4.9 did not work either, error remains. Tips here from the issue also lead to no success. DD completely uninstalled and reinstalled also brings nothing. :(

Diagnostics ID: BB2765D8-51A0-423C-A9E4-01EEBC7657D9/20220614172523

I have a Mac 2019, 16", Intel Core i9 with 32 GB RAM.

**Comment by dspacejs** [comment]
I've started experiencing this issue again:

```sh
$ docker --version
Docker version 20.10.17, build 100c701

$ docker ps
Error response from daemon: dial unix docker.raw.sock: connect: connection refused
```

Docker Desktop is `v4.10.1`.

It looks like it's a slightly different error to last time.

**Comment by leesus** [comment]
Same issue as @dspacejs with the same version.

**Comment by SAMZONG** [comment]
same issue as @dspacejs and @leesus 

- docker-desktop version: 4.10.1 (82475) (latest)
- mbp version: m1 max and macos 12.4

**Comment by thiagoufg** [comment]
Same issue... MacBook Pro 16'' 2020, intel i7, 32 GB RAM, Docker version 20.10.17, build 100c701, Monterey 12.5

**Comment by zcht** [comment]
In the meantime it works again, I tried many things, but only one helped. Here in the [article](https://stackoverflow.com/questions/44346109/how-to-easily-install-and-uninstall-docker-on-macos) we describe how to remove Docker and ALL VMs. After I did that, and last installed the version 4.11.1, it runs again. 

`sudo rm -Rf /Applications/Docker.app
sudo rm -f /usr/local/bin/docker
sudo rm -f /usr/local/bin/docker-machine
sudo rm -f /usr/local/bin/com.docker.cli
sudo rm -f /usr/local/bin/docker-compose
sudo rm -f /usr/local/bin/docker-compose-v1
sudo rm -f /usr/local/bin/docker-credential-desktop
sudo rm -f /usr/local/bin/docker-credential-ecr-login
sudo rm -f /usr/local/bin/docker-credential-osxkeychain
sudo rm -f /usr/local/bin/hub-tool
sudo rm -f /usr/local/bin/hyperkit
sudo rm -f /usr/local/bin/kubectl.docker
sudo rm -f /usr/local/bin/vpnkit
sudo rm -Rf ~/.docker
sudo rm -Rf ~/Library/Containers/com.docker.docker
sudo rm -Rf ~/Library/Application\ Support/Docker\ Desktop
sudo rm -Rf ~/Library/Group\ Containers/group.com.docker
sudo rm -f ~/Library/HTTPStorages/com.docker.docker.binarycookies
sudo rm -f /Library/PrivilegedHelperTools/com.docker.vmnetd
sudo rm -f /Library/LaunchDaemons/com.docker.vmnetd.plist
sudo rm -Rf ~/Library/Logs/Docker\ Desktop
sudo rm -Rf /usr/local/lib/docker
sudo rm -f ~/Library/Preferences/com.docker.docker.plist
sudo rm -Rf ~/Library/Saved\ Application\ State/com.electron.docker-frontend.savedState
sudo rm -f ~/Library/Preferences/com.electron.docker-frontend.plist`

**Comment by adamSplendex** [comment]
@zcht after 4 hours of struggles, this was the only solution, but only 4.11.1 works with this method..i installed all versions back to 4.2 just to try it...but only the above works. 

Thank you and hopefully Docker will sort this out, way too many ppl rely on this.

**Comment by mosijava** [comment]
undortunately for me, it didn't work. I have to revert it to version 4.3.2

**Comment by bluesmoke7** [comment]
I am facing the same problem after upgrading to OSX12.4 with DD version 4.11.1 . tried all the cleanup steps but nothing worked . tried with 4.10.x and 4.9.x but got error saying '“Docker” is damaged and can’t be opened. You should move it to the Bin'. 
the only version that @mosijava mentioned working is the older one - 4.3.2

**Comment by camnv93** [comment]
I have the same problem with 4.11.1
#Macbook pro 14 M1 Os 12.4.
I have tried many different ways but none worked 🥲

**Comment by Shiba-Kar** [comment]
> I am on Mac OSX 12.5 and have the same problem 4.11.1 does not work. I did all the tricks, removed all files and dir and tried the version from the website as well as brew install.
> 
> So how do I go back to an older version?

pls give me the download link to 4.11.1 how to install it using brew?

**Comment by brysonreece** [comment]
I was able to resolve the issue by completely removing Docker using [this script I wrote](https://gist.github.com/brysonreece/a7109c087aeb33aa80c7a560352517b5) and then rebooting my machine. After a re-install everything is working correctly again!

**Comment by ronnyandre** [comment]
Tried your script @brysonreece but unfortunately it didn't work for me :(

macOS Monterey 12.6
Apple M1 Pro

**Comment by ghostandthemachine** [comment]
I'm also on

macOS Monterey 12.6
Apple M1 Pro

And this script did work for me... Thanks @brysonreece. I tried manually uninstalling a few times and re-installing without luck. I think a main difference was everything in desktop was hanging for me so I couldn't even run the uninstaller. Running uninstall with this script did work though... Sorry I can't share any more useful info @ronnyandre

**Comment by brysonreece** [comment]
Yeah I originally went through the same process as @ghostandthemachine, not sure how else to help @ronnyandre; sorry!

**Comment by jv-k** [comment]
With Docker Desktop 4.13.0 on MacOS 12.6  Monterrey, Docker wouldn't open, the process simply freezes before the tray icon appears. I tried also installing with `brew`,  and disabling VPN, they didn't work

Deleting the folders as per below worked, but only with `4.11.1`:
https://stackoverflow.com/questions/44346109/how-to-easily-install-and-uninstall-docker-on-macos/65468254#65468254

**Comment by rauerhans** [comment]
I have the same problem as @jv-k on an Intel Mac from 2019 with 12.6 Monterey. Will try to install 4.11.1 as well now.

**Comment by pranavbadami** [comment]
Hey folks, was stuck on this as well and ended up uninstalling Docker Desktop 4.13.0 from the GUI (Troubleshoot > Uninstall) and then downloaded version[ 4.8.1](https://docs.docker.com/desktop/release-notes/#docker-desktop-481). That at least lets me run `docker run hello-world`. Hopefully helpful to someone! 

I'm on Mac OS 12.0.1 , M1 MBP

**Comment by ronnyandre** [comment]
I successfully managed to install Docker Desktop 4.9.1. Next version (4.10.0) did not work, so I am currently stuck with 4.9.1.

**Comment by jv-k** [comment]
Has anyone who had problems with `4.13.0` and older had any luck with `4.14`, which came out a few days ago? I haven't had a chance to give it a spin, because I'd in the middle of a project and `4.11.1` is not borked yet😅

It's available [here](https://docs.docker.com/desktop/release-notes/#docker-desktop-414).

**Comment by aguedeney** [comment]
Monterey 12.6.1 with 4.14 caused me the Docker Desktop to spin with no way to kill it other than using Activity Monitor.  I fell back to 4.12 which was stable for me. HOWEVER, I had to clean up all of 4.14 installation before reinstalling 4.12, otherwise you'll have various errors occur :(

**Comment by jv-k** [comment]
Oh and now there's macOS Ventura `13.0.1`. I'm going to try it out with `4.14` soon

**Comment by jv-k** [comment]
I just tried `4.14.1` on `macOS Ventura 13.1` — and I still have the exact same problem as before.

Been stuck on since `4.11.0` and no other version works after that. I'm sure it has something to do with Docker Engine.

![image](https://user-images.githubusercontent.com/5808914/210296154-46d19d7f-919c-4d57-9bb5-46d00024d2ed.png)

Is anyone else having the same issue?  What is Docker doing to fix this...?

**Comment by jv-k** [comment]
@dspacejs Totally agreed. Whilst one can argue that it's free and gtfo, but that's not the right way to consider an industry-standard piece of software that you'll come across all the time in this field.

Speaking of which, now recently for some reason (maybe due to upgrading to Ventura..?) files that are mounted inside a container take sometimes minutes to propagate changes from the host — the only solution is to re-start Docker Desktop. It's probably not happening in the latest version of Docker Engine for macOs, but I'd not be able to benefit since I'm stuck on `4.11.0` 🙄

A dedicated Linux box solves all of this, but still, these are pretty bad bugs for such a prominent company + software.
